src/parser/warrior/protection/modules/features/BlockCheck.js

```diff
@@ -55,29 +64,48 @@ class BlockCheck extends Analyzer {
-      } else {
-        this.physicalHitsWithoutShieldBlock += 1;
-        this.physicalDamageWithoutShieldBlock += event.amount + (event.absorbed || 0) + (event.overkill || 0);
-      }
+      event.prot = {
+        shieldBlock: this.selectedCombatant.hasBuff(SPELLS.SHIELD_BLOCK_BUFF.id),//ability to look what important buffs they had retro actively
+        bloster: (this.bolster && this.selectedCombatant.hasBuff(SPELLS.LAST_STAND.id)),
+      };
+      this.listOfEvents.push(event);
     }
   }
 
   on_fightend() {
+    const blockableSet = new Set();//this is master list of all BLOCKED events in the fight
+    blockableSet.add(1);//make it so if they never hit sb we still get data from the melees they take
+    this.shieldBlock.shieldBlocksDefensive.forEach(function(block){
+      block.eventName.forEach(function(blockedAbility){
+        blockableSet.add(blockedAbility);//just go through one set to another
+      });
+    });
+
+    const that = this;
+    this.listOfEvents.forEach(function(event){
+      if(blockableSet.has(event.ability.guid)){//if it ain't been blocked over the whole fight it prob aint blockable
+        if (event.prot.shieldBlock || event.prot.bloster) {//they got block up when it happened?
+          that.physicalHitsWithBlock += 1;
+          that.physicalDamageWithBlock += event.amount + (event.absorbed || 0) + (event.overkill || 0);
+        } else {
+          that.physicalHitsWithoutBlock += 1;
+          that.physicalDamageWithoutBlock += event.amount + (event.absorbed || 0) + (event.overkill || 0);
+        }
+      }
+    });
+
     if (debug) {
-      console.log(`Hits with block spell up ${this.physicalHitsWithShieldBlock}`);
-      console.log(`Damage with block spell up ${this.physicalDamageWithShieldBlock}`);
-      console.log(`Hits without block spell up ${this.physicalHitsWithoutShieldBlock}`);
-      console.log(`Damage without block spell up ${this.physicalDamageWithoutShieldBlock}`);
-      console.log(`Total physical ${this.physicalDamageWithoutShieldBlock}${this.physicalDamageWithShieldBlock}`);
+      console.log(`Hits with block spell up ${this.physicalHitsWithBlock}`);
+      console.log(`Damage with block spell up ${this.physicalDamageWithBlock}`);
+      console.log(`Hits without block spell up ${this.physicalHitsWithoutBlock}`);
+      console.log(`Damage without block spell up ${this.physicalDamageWithoutBlock}`);
+      console.log(`Total physical ${this.physicalDamageWithoutBlock}${this.physicalDamageWithBlock}`);
     }
   }
 
   get suggestionThresholds() {//was in here before but is/was never used and appears to be very high requirements that are unreasonable maybe lower and add laster?
     return {
-      actual: this.physicalDamageWithShieldBlock / (this.physicalDamageWithShieldBlock + this.physicalDamageWithoutShieldBlock),
+      actual: this.physicalDamageWithBlock / (this.physicalDamageWithBlock + this.physicalDamageWithoutBlock),
```

Comment:
this should probably just be % of hits blocked, with special handling for blockable tankbusters. i get the idea (threshold on % of damage where you had block up), but a) this is currently incorrect (to do that correctly it should be `sum of raw damage from blocked hits / sum of raw damage`), and b) its possible to do well on that purely by luck even with bad SB uptime.